### PR TITLE
RDKTV-11503 : Gdial Crash test

### PR DESCRIPTION
**Pull request by akhilsarada**
Reason for change: gdial graceful closing.
Test Procedure: Refer JIRA
Risks: Low
Signed-off-by: Akhil Baby Sarada <Akhil_Sarada@comcast.com>

### CODE REVIEW

File: server/main.c [hunk 1]
```diff
@@ -356,6 +356,34 @@ int main(int argc, char *argv[]) {
    */
   loop_ = g_main_loop_new (NULL, TRUE);
   g_main_loop_run (loop_);
+  gdial_rest_server_unregister_all_apps(dial_rest_server);
```
Comment (apatel859) [inline server/main.c:359]: I was suggesting to call server_activation_handler(0, ""); from gdial_quit_thread

Comment (apatel859) [inline server/main.c:359]: no need to unregister apps as anyway process is restarting deactivate will make sure we dont handle any http request

Comment (akhilsarada) [reply server/main.c]: Updated as suggested

Comment (akhilsarada) [reply server/main.c]: Provided sleep after server_activation_handler(0, "");  to settle . Removing application status checks

File: server/main.c [hunk 2]
```diff
@@ -178,6 +178,8 @@ static void gdial_http_server_throttle_callback(SoupServer *server,
 static void gdial_quit_thread(int signum)
 {
   g_print("Exiting DIAL Server thread %d \r\n",signum);
+  server_activation_handler(0, "");
+  sleep(3);
```
Comment (apatel859) [inline server/main.c:182]: Add a comment here
// Sleeping 3 sec to allow existing request to be finished processing.

**Review by apatel859 — APPROVED** [review]

**Pull request merged**